pkg/expectations/expectations.go

```diff
@@ -0,0 +1,347 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/* TODO:
+Taken from https://github.com/kubernetes/kubernetes/blob/1889a6ef52eb18b08e24843577c5b9d3b9a65daa/pkg/controller/controller_utils.go
+As soon as expectations become available in client-go or apimachinery, delete this and switch.
+*/
+
```

Comment:
Any way we can vendor in the kubevirt code for this?

Reply:
it would create a circular dependency

kubevirt depends on cdi for datavolume client, cdi would then depend on kubevirt for expectations.